test/Events.test.js

```diff
@@ -0,0 +1,328 @@
+const expect = require("chai").expect;
+const EventEmitter = require("events");
+const sinon = require("sinon");
+const { GameState } = require("../entities/GameState");
+const { createPlayer } = require("./testutils");
+const { PropertyService } = require("../entities/PropertyService");
+
+describe("Events", () => {
+  let gameState;
+  let mockUI;
+  let eventBusEmitter;
+  let EVENTS;
+
+  beforeEach(() => {
+    gameState = new GameState();
+    gameState.currentPlayerActions = {
+      END_TURN: {
+        execute: () => true,
+      },
+    };
+    mockUI = {
+      startGame: () => true,
+      startTurn: (player) => false,
+      displayAvailableActions: (actions) => false,
+      prompt: () => true,
+      endTurn: () => true,
+      rollingDice: () => true,
+      rollDiceDisplay: (shouldDisplay) => true,
+      payFineDisplay: (shouldDisplay) => true,
+      endTurnDisplay: (shouldDisplay) => true,
+      diceRollResults: (roll1, roll2) => true,
+      rollNormalDice: () => true,
+      rollJailDice: () => true,
+      caughtSpeeding: () => true,
+      playerMovement: (position) => true,
+      payFine: () => true,
+      passGo: () => true,
+      jail: () => true,
+    };
+    eventBusEmitter = new EventEmitter();
+    EVENTS = require("../entities/Events")(eventBusEmitter, mockUI, gameState);
+    gameState.players = [createPlayer({ name: "player1" })];
+  });
+  afterEach(() => {
+    // Restore the default sandbox here
+    sinon.restore();
+  });
+  it("constructor should return object with associated event listeners", () => {
```

Comment:
I think this is kind of brittle - could we drop the `length` check and just for-in to check `typeof x === "function"` ?
The advantage of using the Function Names is to ensure we have them, but I think that that belongs in at the integration test level as it is more about a contract between entities using said events rather than representing the nature of the Event Object as a unit